acme/acme.go

```diff
@@ -331,6 +334,14 @@ func (a *ACME) CreateLocalConfig(tlsConfig *tls.Config, checkOnDemandDomain func
 func (a *ACME) getCertificate(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
 	domain := types.CanonicalDomain(clientHello.ServerName)
 	account := a.store.Get().(*Account)
+	//use regex to test for wildcard certs that might have been added into TLSConfig
+	for k := range a.TLSConfig.NameToCertificate {
+		selector := "^" + strings.Replace(k, "*.", ".*\\.?", -1) + "$"
```

Comment:
@dtomcej Will this properly **not** match subdomains of a wildcard?
E,g, If I have a certificate for `*.mysite.com`, then `foo.mysite.com` should match but `bar.foo.mysite.com` should not match.
Should it be `strings.Replace(k, "*.", "[^.]*\\.?", -1)` instead?